File: Benchmarks/GibberishComparison.cs
using System;
using System.IO;
using System.Text;
using System.Text.Patterns;
using System.Text.RegularExpressions;
using BenchmarkDotNet.Attributes;

namespace Benchmarks {
	[ClrJob, CoreJob, CoreRtJob]
	[MemoryDiagnoser]
	public class GibberishComparison {

		public String gibberish = Gibberish.Generate(4);

		public String badGibberish = Gibberish.Generate(4, Bad: true);

		readonly Pattern pattern = +(+Pattern.Check("Letter", (Char) => 'a' <= Char && Char <= 'z') | +(Pattern)' ') & Source.End;

		readonly Regex regex = new Regex(@"(?:[a-z]+| +)+$", RegexOptions.Singleline);

		[Benchmark]
		public void PatternGood() {
			Source Source = new Source(gibberish);
			pattern.Consume(ref Source);
		}

		[Benchmark]
		public void PatternBad() {
			Source Source = new Source(badGibberish);
			pattern.Consume(ref Source);
		}

		[Benchmark]
		public void RegexGood() {
			regex.Match(gibberish);
		}

		[Benchmark]
		public void RegexBad() {
			regex.Match(badGibberish);
		}
	}
}

File: Benchmarks.FSharp/GibberishComparison.fs
namespace Benchmarks

open System
open System.IO
open System.Text
open System.Text.Patterns
open FParsec
open BenchmarkDotNet.Attributes

[<ClrJob>]
[<MemoryDiagnoser>]
type GibberishComparison() =
    let patternWord:Pattern = +(check "letter" (new Func<Char,Boolean>(fun (char) -> 'a' <= char && char <= 'z')))
    let patternSpace:Pattern = +' '
    let pattern:Pattern = +(patternWord || patternSpace) >> Pattern.EndOfSource

    let parsecWord = many1Chars (anyOf "abcdefghijklmnopqrstuvwxyz")
    let parsecSpace = many1Chars (pchar ' ')
    let parsec = many(parsecWord <|> parsecSpace) .>>. eof

    let gibberish = Gibberish.Generate(4)
    let badGibberish = Gibberish.Generate(4, Bad=true)

    [<Benchmark>]
    member this.PatternGood() =
        let mutable source = new Source(gibberish)
        let result = pattern.Consume(&source)
        ()

    [<Benchmark>]
    member this.PatternBad() =
        let mutable source = new Source(badGibberish)
        let result = pattern.Consume(&source)
        ()

    [<Benchmark>]
    member this.FParsecGood() =
        let result = run parsec gibberish
        ()

    [<Benchmark>]
    member this.FParsecBad() =
        let result = run parsec badGibberish
        ()